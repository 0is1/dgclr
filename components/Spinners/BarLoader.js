import React from 'react';
import BarLoader from 'react-spinners/BarLoader';
import { Box, Flex } from 'rebass';
import colors from 'components/colors';

const Spinner = () => (
  <Flex alignItems="center" justifyContent="center">
    <Box>
      <BarLoader className="loading" sizeUnit="%" width={80} height={8} color={colors.info} loading />
    </Box>
  </Flex>
);

export default Spinner;
