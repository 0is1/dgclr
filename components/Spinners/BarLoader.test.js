/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { BarLoader as BarLoaderSpinner } from 'react-spinners';
import { Box } from 'rebass';
import BarLoader from './BarLoader';

describe('Layout component', () => {
  it('Match snapshot', () => {
    const component = renderer.create(<BarLoader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Renders correctly', () => {
    const enzymeWrapper = shallow(<BarLoader />);
    expect(enzymeWrapper).toBeDefined();
    expect(enzymeWrapper.find(Box).length).toEqual(1);
    expect(enzymeWrapper.find(BarLoaderSpinner).length).toEqual(1);
  });
});
