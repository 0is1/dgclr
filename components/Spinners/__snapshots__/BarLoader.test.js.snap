// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Match snapshot 1`] = `
<Flex
  alignItems="center"
  extend={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {},
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "baseHash": 400283751,
        "componentId": "sc-AxjAm",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
        ],
        "staticRulesId": "",
      },
      "displayName": "Box",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "alignSelf": [Function],
        "bg": [Function],
        "color": [Function],
        "flex": [Function],
        "fontSize": [Function],
        "m": [Function],
        "mb": [Function],
        "ml": [Function],
        "mr": [Function],
        "mt": [Function],
        "mx": [Function],
        "my": [Function],
        "order": [Function],
        "p": [Function],
        "pb": [Function],
        "pl": [Function],
        "pr": [Function],
        "pt": [Function],
        "px": [Function],
        "py": [Function],
        "width": [Function],
      },
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "sc-AxjAm",
      "systemComponent": true,
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  justifyContent="center"
>
  <Box>
    <Loader
      className="loading"
      color="#6173fb"
      css=""
      height={8}
      loading={true}
      sizeUnit="%"
      width={80}
    />
  </Box>
</Flex>
`;
